internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1214,6 +1214,13 @@ func (r *ClusterDeploymentsReconciler) TransformClusterToDay2(
 	if err != nil {
 		log.WithError(err).Errorf("failed to transform cluster %s into day2 cluster", cluster.ID.String())
 	}
+	// Set custom Ignition endpoint so that day2 workers would join using HTTPS endpoint
+	// ensureMCSCert would add ignition override with MCS secret
+	ignitionEndpoint := fmt.Sprintf("https://api-int.%s.%s:22623/config/%s", cluster.Name, cluster.BaseDNSDomain, models.HostRoleWorker)
+	clusterInstall.Spec.IgnitionEndpoint = &hiveext.IgnitionEndpoint{
```

Comment:
This can conflict with the capi-provider-agent or any other user that override the default ignition endpoint
e.g.
https://github.com/openshift/cluster-api-provider-agent/blob/7266349eec6bf9f54260e1e01921a9bfeacf71b7/controllers/agentcluster_controller.go#L285

Reply:
we can validate if ignitionEndpoint was set then don't change it

Reply:
Good idea, wrapped it in `if`